shop/src/main/java/ro/msg/learning/shop/service/LocationService.java

```diff
@@ -0,0 +1,38 @@
+package ro.msg.learning.shop.service;
+
+import org.springframework.stereotype.Service;
+import ro.msg.learning.shop.dto.save.SaveLocationDto;
+import ro.msg.learning.shop.model.Location;
+import ro.msg.learning.shop.repository.LocationRepository;
+
+import javax.transaction.Transactional;
+import java.util.List;
+import java.util.Optional;
+
+@Service
+public class LocationService {
+    private final LocationRepository locations;
+
+    public LocationService(LocationRepository locations) {
+        this.locations = locations;
+    }
+
+    public Location saveLocation(SaveLocationDto location) {
+        return locations.save(location.toLocation());

```

Comment:
it is generally preffered that the service have nothing to do with dtos. therefore we can move this mapping in the controller and return from there a mapped version of the entity we play with here.
Plenty of reasons for that. Ex: your service layer and the layers below could be used in the future as a "base" for a Romanian specific version of your app, where to the Dto we need to add romanian specific logic etc. And at the same time by a German specific version...etc. The point is your service layer is the base ice cream and should be able to be used with any kind of toppings 🍨